dev/tools/create_api_docs.dart

```diff
@@ -0,0 +1,1107 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math' as math;
+
+import 'package:archive/archive_io.dart';
+import 'package:args/args.dart';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:http/http.dart' as http;
+import 'package:intl/intl.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+import 'package:pub_semver/pub_semver.dart';
+
+import 'dartdoc_checker.dart';
+import 'examples_smoke_test.dart';
+
+FileSystem filesystem = const LocalFileSystem();
```

Comment:
nit

```suggestion
const FileSystem filesystem = LocalFileSystem();
```

Reply:
No, that was on purpose: I wanted tests to be able to set that global to a memory filesystem (even though I don't actually use that capability yet).

Reply:
Actually, in talking to Chris, I think he's right: if I write more tests for this, I'd be testing the main classes, not the main function itself.

Reply:
I've seen my share of test flakes from this pattern :)